lib/matplotlib/tests/test_figure.py

```diff
@@ -234,3 +236,38 @@ def test_figaspect():
     assert h / w == 0.5
     w, h = plt.figaspect(np.zeros((2, 2)))
     assert h / w == 1
+
+def test_autofmt_xdate():
+    for which in [None, 'both', 'major', 'minor']:
+        
+        date = ['3 Jan 2013', '4 Jan 2013', '5 Jan 2013', '6 Jan 2013', '7 Jan 2013',
+        '8 Jan 2013', '9 Jan 2013', '10 Jan 2013', '11 Jan 2013', '12 Jan 2013',
+        '13 Jan 2013', '14 Jan 2013']
+
+        time = ['16:44:00', '16:45:00', '16:46:00', '16:47:00', '16:48:00', '16:49:00',
+        '16:51:00', '16:52:00', '16:53:00', '16:55:00', '16:56:00', '16:57:00']
+
+        angle = 60
+        minors = [1,2,3,4,5,6,7]
+        
+        x = mdates.datestr2num(date)
+	y = mdates.datestr2num(time)
```

Comment:
It looks like you have a mix of tabs and spaces, can you please make them all spaces (many editors have tools to help with this, and a minimum find and replace `\t' -> '    '.